crypto/param_build.c

```diff
@@ -159,6 +159,33 @@ int ossl_param_bld_push_BN(OSSL_PARAM_BLD *bld, const char *key,
     return 1;
 }
 
+int ossl_param_bld_push_BN_pad(OSSL_PARAM_BLD *bld, const char *key,
+                               const BIGNUM *bn, size_t sz)
+{
+    int n, secure = 0;
+    OSSL_PARAM_BLD_DEF *pd;
+
+    if (bn != NULL) {
```

Comment:
What are the semantics of pushing a NULL bn?  Isn't that an error?

Reply:
It's perfectly fine. For example, dh keys may have a NULL Q, and even though silly to pass that, it makes it easier to call this function without surrounding it with extre checks (that you might forget to do).

It may also be that when getting parameters, BNs can get allocated automatically as needed on the owning side.

Reply:
ok, tnx.